core/modules/admin_bar/admin_bar.theme.inc

```diff
@@ -22,6 +22,16 @@ function theme_admin_bar_icon_page($variables) {
  return t('This page');
 }
 
+/**
+* Renders a language icon to display in the administration bar.
+*
+* @ingroup themeable
+*/
+function theme_admin_bar_icon_locale($variables) {
+  global $language;
+  return $language->name;
```

Comment:
I suggest changing this to:

```suggestion
  return $language->native;
```

To show the native spelling if the language in the switcher
<img width="279" alt="Screen Shot 2023-04-15 at 7 40 31 AM" src="https://user-images.githubusercontent.com/9938978/232227583-1d13d3db-9a53-48f0-b779-db6ed4d5e9c0.png">